Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -0,0 +1,249 @@
+        try {
+            Timber.tag(TAG).v("Update the risk level with $riskLevel")
+            val lastCalculatedRiskLevelScoreForRollback =
+                RiskLevelRepository.getLastCalculatedScore()
+            updateRiskLevelScore(riskLevel)
+            rollbackItems.add {
+                updateRiskLevelScore(lastCalculatedRiskLevelScoreForRollback)
+            }
+
+            // risk level calculation date update
+            val lastCalculatedRiskLevelDate = LocalData.lastTimeRiskLevelCalculation()
+            LocalData.lastTimeRiskLevelCalculation(time)
+            rollbackItems.add {
+                LocalData.lastTimeRiskLevelCalculation(lastCalculatedRiskLevelDate)
+            }
+        } catch (error: Exception) {
+            Timber.tag(TAG).e(error, "Updating the RiskLevelRepository failed.")
+
+            try {
+                Timber.tag(TAG).d("Initiate Rollback")
+                for (rollbackItem: RollbackItem in rollbackItems) rollbackItem.invoke()
+            } catch (rollbackException: Exception) {
+                Timber.tag(TAG).e(rollbackException, "RiskLevelRepository rollback failed.")
+            }
+
+            throw error
+        }
+    }
+
+    override fun calculationNotPossibleBecauseOfOutdatedResults(): Boolean {
+        // if the last calculation is longer in the past as the defined threshold we return the stale state
+        val timeSinceLastDiagnosisKeyFetchFromServer =
+            TimeVariables.getTimeSinceLastDiagnosisKeyFetchFromServer()
+                ?: throw RiskLevelCalculationException(
+                    IllegalArgumentException(
+                        "Time since last exposure calculation is null"
+                    )
+                )
+        /** we only return outdated risk level if the threshold is reached AND the active tracing time is above the
+        defined threshold because [UNKNOWN_RISK_INITIAL] overrules [UNKNOWN_RISK_OUTDATED_RESULTS] */
+        return timeSinceLastDiagnosisKeyFetchFromServer.millisecondsToHours() >
+            TimeVariables.getMaxStaleExposureRiskRange() && isActiveTracingTimeAboveThreshold()
+    }
+
+    override fun calculationNotPossibleBecauseNoKeys() =
+        (TimeVariables.getLastTimeDiagnosisKeysFromServerFetch() == null).also {
+            if (it) {
+                Timber.tag(TAG)
+                    .v("No last time diagnosis keys from server fetch timestamp was found")
+            }
+        }
+
+    override suspend fun calculationNotPossibleBecauseTracingIsOff() =
+        // this applies if tracing is not activated
+        !enfClient.isTracingEnabled.first()
+
+    override suspend fun isIncreasedRisk(): Boolean {
+        val lastExposureSummary = getNewExposureSummary()
+        val appConfiguration = appConfigProvider.getAppConfig()
+        Timber.tag(TAG).v("Retrieved configuration from backend")
+        // custom attenuation parameters to weight the attenuation
```

Comment:
Weigh, not weight.

Reply:
Should we then also rename the field `attenuationParameters.weights`?

Reply:
I don't think so. Weight is used for defining the "heaviness" of something, and the verb "to weigh" is used to find the weight of something. The attenuationParameters indeed have weights. Sorry I don't mean to sound like the grammar police :D

Reply:
I didn't know that! Thank you so much!